backend-api/src/functions/asyncActiveSession/asyncActiveSessionHandler.ts

```diff
@@ -26,55 +26,59 @@ export async function lambdaHandlerConstructor(
     event.headers["Authorization"] ?? event.headers["authorization"],
   );
   if (authorizationHeaderResult.isError) {
-    logger.log("AUTHENTICATION_HEADER_INVALID", {
+    logger.log("INVALID_AUTHENTICATION_HEADER", {
       errorMessage: authorizationHeaderResult.value.errorMessage,
     });
     return unauthorizedResponse;
   }
-
   const serviceTokenJwe = authorizationHeaderResult.value;
 
   const decryptResult = await dependencies
     .jweDecrypter(config.ENCRYPTION_KEY_ARN)
     .decrypt(serviceTokenJwe);
-
   if (decryptResult.isError) {
-    logger.log("JWE_DECRYPTION_ERROR", {
+    if (decryptResult.value.errorCategory === "CLIENT_ERROR") {
+      logger.log("JWE_DECRYPTION_ERROR", {
+        errorMessage: decryptResult.value.errorMessage,
+      });
+      return badRequestResponse("Failed to decrypt service token");
+    }
+    logger.log("INTERNAL_SERVER_ERROR", {
       errorMessage: decryptResult.value.errorMessage,
     });
-    return badRequestResponse("Failed decrypting service token JWE");
+    return serverErrorResponse;
   }
-
   const serviceTokenJwt = decryptResult.value;
+  console.log(serviceTokenJwt);
 
-  const tokenServiceDependencies = dependencies.tokenServiceDependencies;
-  const tokenService = dependencies.tokenService(tokenServiceDependencies);
-  const getSubFromTokenResult = await tokenService.getSubFromToken(
-    config.STS_JWKS_ENDPOINT,
+  const tokenService = dependencies.tokenService(config.STS_JWKS_ENDPOINT);
+  const validateServiceTokenResult = await tokenService.validateServiceToken(
+    serviceTokenJwt,
     {
```

Comment:
I thought that since we don't intend to make these values environment variables then there is no point in passing them through all the way from the handler to the `sendHttpRequest` function, so I've moved this object to `publicKeyGetter.ts`